Shift responsibility for keeping Date header to ResponseHeaderBag

mpdude opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

This is an improvement over #22036. It shifts responsibility for preserving a `Date` header to the `ResponseHeaderBag`.

We already have similar logic there for the `Cache-Control` header.

mpdude committed:
Guard Date header in ResponseHeaderBag instead of Response

fabpot commented:
I would do that in master as there is no bug fix here.

mpdude commented:
@fabpot Fine for me. Can you do this when merging, or do you need a new PR?

fabpot commented:
You can change the target branch on the PR directly.

mpdude commented:
Must be a new GH feature or it has always been so well hidden I missed it.

Changed.

ro0NL commented on src/Symfony/Component/HttpFoundation/Tests/ResponseHeaderBagTest.php:
```diff
@@ -12,56 +12,33 @@
-            ),
-            array(
-                array('ETag' => 'xyzzy'),
-                array('ETag' => array('xyzzy'), 'Cache-Control' => array('private, must-revalidate')),
```
Im not sure you can simply remove the cache-control expectations, perhaps re-add it in a separate test. Which is probably best in the long run.

mpdude replied:
If I am not mistaken, there are other tests for them. In this test method that was only noise AFAICT, but please correct me if I'm wrong

ro0NL replied:
You're right :+1: same situation is covered in `testCacheControlHeader`

mpdude replied:
Thanks!

mpdude commented:
@fabpot Does that mean you've postponed the merge from 3.3 to 3.4 :-(?

mpdude commented:
@fabpot Anything left here I could do?

fabpot reviewed (approved):
Can you rebase before I merge to get rid of the merge commits? Thanks.

fabpot commented:
If you could also change the target to 3.4, that would be wonderful.

mpdude commented:
@fabpot changed base to 3.4 – did I get you right?

Also, at least in the web interface, GitHub shows no merge conflicts...?

fabpot commented:
3.4 👍 
Can you still rebase please? We don't merge PRs with merge commits.

mpdude commented:
Hm, rebasing and fixing up a branch that I had previously rebased in the GH web interface, including merge conflicts, really challenged my git skills... hope it's allright now.

fabpot commented:
Apparently, tests are broken on deps low.

mpdude committed:
Fix tests

The headers returned now also include the "Date", but adding it to the expected outcome would add additional noise.

So instead I chose to test that only the given headers would remain present, preserving case.

The Cache-Control was not important here, we have other tests covering that aspect.

mpdude commented:
Messed up the test during merge conflict resolution, should be fixed now.

fabpot commented:
Thank you @mpdude.
